contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,76 @@
+* Incorporate priority in pod scheduling logic.
+
+## Background 
+
+Since we already have [priority filed in Pod spec](https://github.com/kubernetes/kubernetes/pull/45610), 
+Pods can now be classified into different priority classes. So it is nature to make ResourceQuota apply to Pods based on its priority class. In order to implement this, we need to add a priority class name filed to ResourceQuota definition.
+
+## Overview 
+
+This design doc introduces the motivation of adding priority filed in ResourceQuota and
+how this priority impacts ResourceQuta applying to Pod. 
+
+## Detailed Design 
+
+### Expected behavior of ResourceQuota apply to pod 
+
+The design will be like:
+
+```go
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+  // Hard is the set of desired hard limits for each named resource
+  // +optional
+  Hard ResourceList
+  // A collection of filters that must match each object tracked by a quota.
+  // If not specified, the quota matches all objects.
+  // +optional
+  Scopes []ResourceQuotaScope
+  // If specified, indicates this ResourceQuota applies to that priority class
+  // If not specified, this ResourceQuota applies to all priority classes.
+  // +optional
+  PriorityClassName string
+}
+```
+PriorityClassName in ResourceQuota is matched against PriorityClassName in the Pod, including the case where PriorityClassName is not set. In detail:
+
+* One could expect when `PriorityClassName` field is set, it indicates that this quota applies to Pod with same priority class. 
+* If this field is not present, it indicates this quota match to pods that do not have a PriorityClassName (Pods' integer priorities would be ignored).
+
+It is worth noting that quota is assigned per namespace, not per user, in Kubernetes. Adding priority class name to ResourceQuota will not change anything with respect to the scope of ResourceQuota. In other words, when priority class name is defined for a ResourceQuota object, the quota is applied to all pods at that priority class in the specified namespace.
+
+Unlike pod spec, we do not have `Priority *int32` in `ResourceQuta` spec.  It's because we allow ResourceQuota with empty PriorityClassName and such a ResourceQuota matches pods that do not have a PriorityClassName. We think this design is more straightforward for end user.
+
+### Expected behavior of ResourceQuota admission controller
+
+This part is about how to make admission controller work with priority aware `ResourceQuota`.
+
+* ResourceQuota admission controller should reject creation or update of ResourceQuota object if new ResourceQuota names a PriorityClass that does not exist, i.e. is not a pre-defined or user-specified PriorityClass (but empty is allowed).
+* ResourceQuota admission controller should reject creation of ResourceQuota object if another ResourceQuota object with same PriorityClass already exists, or if a ResourceQuota object with no PriorityClass exists.
```

Comment:
With the recent change of plans, the admission controller should no longer reject creation of a resource quota object with a priorityclass when a different one with no priority class exists.